Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,519 @@
+                                                    { CFSTR("EXTRALIGHT"), DWRITE_FONT_WEIGHT_EXTRA_LIGHT },
+                                                    { CFSTR("ULTRALIGHT"), DWRITE_FONT_WEIGHT_ULTRA_LIGHT },
+                                                    { CFSTR("LIGHT"), DWRITE_FONT_WEIGHT_LIGHT },
+                                                    { CFSTR("SEMILIGHT"), DWRITE_FONT_WEIGHT_SEMI_LIGHT },
+                                                    { CFSTR("NORMAL"), DWRITE_FONT_WEIGHT_NORMAL },
+                                                    { CFSTR("REGULAR"), DWRITE_FONT_WEIGHT_REGULAR },
+                                                    { CFSTR("MEDIUM"), DWRITE_FONT_WEIGHT_REGULAR },
+                                                    { CFSTR("DEMIBOLD"), DWRITE_FONT_WEIGHT_DEMI_BOLD },
+                                                    { CFSTR("DEMI"), DWRITE_FONT_WEIGHT_DEMI_BOLD },
+                                                    { CFSTR("SEMIBOLD"), DWRITE_FONT_WEIGHT_SEMI_BOLD },
+                                                    { CFSTR("BOLD"), DWRITE_FONT_WEIGHT_BOLD },
+                                                    { CFSTR("EXTRABOLD"), DWRITE_FONT_WEIGHT_EXTRA_BOLD },
+                                                    { CFSTR("ULTRABOLD"), DWRITE_FONT_WEIGHT_ULTRA_BOLD },
+                                                    { CFSTR("BLACK"), DWRITE_FONT_WEIGHT_BLACK },
+                                                    { CFSTR("HEAVY"), DWRITE_FONT_WEIGHT_HEAVY },
+                                                    { CFSTR("EXTRABLACK"), DWRITE_FONT_WEIGHT_EXTRA_BLACK },
+                                                    { CFSTR("ULTRABLACK"), DWRITE_FONT_WEIGHT_ULTRA_BLACK } };
+
+/**
+ * Helper that parses a font name, and returns appropriate weight, stretch, style, and family name values
+ */
+_DWriteFontProperties _DWriteGetFontPropertiesFromName(CFStringRef fontName) {
+    // Set some defaults for when weight/stretch/style are not mentioned in the name
+    _DWriteFontProperties ret{ DWRITE_FONT_WEIGHT_NORMAL,
+                               DWRITE_FONT_STRETCH_NORMAL,
+                               DWRITE_FONT_STYLE_NORMAL,
+                               _DWriteGetFamilyNameForFontName(fontName) };
+
+    // Relationship of family name -> font name not always consistent
+    // Usually, properties are added to the end (eg: Arial -> Arial Narrow Bold)
+    // However, this is not always the case (eg: Eras ITC -> Eras Bold ITC)
+    // In addition, some fonts with properties are occasionally placed into their own family (eg: Segoe WP SemiLight -> Segoe WP SemiLight)
+    // Try to be more prudent about these edge cases, by looking only at the difference between the font name and family name
+    woc::unique_cf<CFArrayRef> fontNameTokens(CFStringCreateArrayBySeparatingStrings(kCFAllocatorDefault, fontName, CFSTR(" ")));
```

Comment:
This would break for a font with a space in its name - "Times New Roman", "Courier New"